src/PortingAssistant.Client.Telemetry/Uploader.cs

```diff
@@ -12,130 +10,251 @@
+            }
+            return logName;
+        }
+
+        public bool Upload(IEnumerable<string> fileEntries)
         {
             try
             {
-                var fileEntries = Directory.GetFiles(teleConfig.LogsPath)
-                    .Where(f =>
-                        Path.GetFileName(f)
-                            .StartsWith("portingAssistant-client-cli") &&
-                        teleConfig.Suffix.ToArray().Any(x => f.EndsWith(x))
-                    ).ToList();
-
-                var lastReadTokenFile = Path.Combine(teleConfig.LogsPath, "lastToken.json");
-                var fileLineNumberMap = new Dictionary<string, int>();
-                if (File.Exists(lastReadTokenFile))
+                foreach (var file in fileEntries)
                 {
-                    fileLineNumberMap = JsonConvert.DeserializeObject<Dictionary<string, int>>(File.ReadAllText(lastReadTokenFile));
+                    var logName = GetLogName(file);
+                    if (string.IsNullOrEmpty(logName))
+                    {
+                        continue;
+                    }
+                    if (_client != null)
+                    {
+                        UploadFile(file, logName);
+                    }
                 }
-                foreach (var file in fileEntries)
+                if (_updatedFileLineNumberMap.Any())
+                {
+                    UpdateFileLineMapJson();
+                }
+                return true;
+            }
+            catch (Exception ex)
+            {
+                ErrorList.Add("UploadFilesError", ex);
+                return false;
+            }
+        }
+
+        public void CleanupLogFolder()
+        {
+            try
+            {
+                long folderSizeLimit = _configuration.LogsFolderSizeLimit != 0
+                    ? _configuration.LogsFolderSizeLimit
+                    : 250000000;
+                long currentSize = 0;
+                ReadFileLineMap();
+                var logsDirectory = new DirectoryInfo(_configuration.LogsPath);
+                bool updateLastTokenJson = false;
+                foreach (var file in logsDirectory
+                             .GetFiles()
+                             .Where(file => _configuration.Suffix.ToArray().Any(file.Name.EndsWith))
+                             .OrderByDescending(f => f.LastWriteTime))
                 {
-                    var logName = Path.GetFileNameWithoutExtension(file);
-                    var fileExtension = Path.GetExtension(file);
-                    if (fileExtension == ".log")
+                    if (currentSize <= folderSizeLimit)
                     {
-                        logName = $"{logName}-logs";
+                        currentSize += file.Length;
+                        continue;
                     }
-                    else if (fileExtension == ".metrics")
+
+                    File.Delete(file.FullName);
```

Comment:
Always assume File.Delete might fail. What would happen in this case? Do we want execution to continue or do we need to terminate the loop?

Reply:
Good point, probably catch and continue execution. I will make a change